exporter/fileexporter/file_exporter.go

```diff
@@ -46,6 +47,8 @@ var logsMarshalers = map[string]plog.Marshaler{
 	formatTypeProto: plog.NewProtoMarshaler(),
 }
 
+var encoder, _ = zstd.NewWriter(nil)
```

Comment:
Are you sure we can use a single instance of the encoder?

It's possible that a single instance of this exporter will receive data from many pipelines. Would this be a problem?

Reply:
The [README](https://github.com/klauspost/compress/tree/master/zstd) indicates this function `EncodeAll` can be called concurrently. Each call will only run on a same goroutine as the caller.
At the same time, I added unit test `TestConcurrentlyCompress` to check whether using the `Encoder` for individual blocks concurrently is safe.
So, Can we reuse the encoder?

Reply:
I will save the encoder on the exporter struct,just in case.